WC_Layers/Source_Mesh_Reader.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Sep 30 15:16:44 2024
+
+@author: Anupama Rajendra
+"""
+
+import h5py
+import numpy as np
+
+source_density_name = 'source_density'
+
+# Open the HDF5 files
+File_1 = h5py.File("source_mesh_1.h5m", 'r')
+File_2 = h5py.File("source_mesh_2.h5m", 'r')
+Files = [File_1, File_2]
+
+def extract_source_data(file_list):
+    '''
+    Identifies the location of the source density dataset within each mesh file.
+    '''
+    sd_list = []
+    for file in file_list:
+        # Extract data from the current file
+        tstt = file['tstt']
+        elements = tstt['elements']
+        tet_four = elements['Tet4']
+        tags = tet_four['tags']
+        sd = tags['source_density'][:]
+        sd_list.append(sd)
+    return sd_list
+    
+def arrange_source_strengths(sd_list, photon_groups):
+    '''
+    Writes individual mesh element strengths (separated by energy group) to density_list, 

```

Comment:
It doesn't look like this method "writes" anything...  It may not be useful to describe the steps that are shown in the code, but summarize the overall purpose.

Reply:
I think this function could be deleted entirely - the source strengths and the relevant sums are now being accessed directly through extract_source_data in the photon transport code